toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1421,17 +1437,24 @@ func InstallBootloader(installChroot *safechroot.Chroot, encryptEnabled bool, bo
 // as part of a general "root" partition is assumed to have been done already.
 func installLegacyBootloader(installChroot *safechroot.Chroot, bootDevPath string) (err error) {
 	const (
-		squashErrors = false
+		squashErrors     = false
+		bootDir          = "/boot"
+		bootDirArg       = "--boot-directory"
+		targetArg        = "--target"
+		legacyTargetArch = "i386-pc"
+		grub2BootDir     = "/boot/grub2"
 	)
 
-	// Since we do not have grub2-pc installed in the setup environment, we need to generate the legacy grub bootloader
-	// inside of the install environment. This assumes the install environment has the grub2-pc package installed
-	err = installChroot.UnsafeRun(func() (err error) {
-		err = shell.ExecuteLive(squashErrors, "grub2-install", "--target=i386-pc", "--boot-directory=/boot", bootDevPath)
-		err = shell.ExecuteLive(squashErrors, "chmod", "-R", "go-rwx", "/boot/grub2/")
-		return
-	})
+	grub2InstallTargetArg := fmt.Sprintf("%s=%s", targetArg, legacyTargetArch)
+	installBootDir := filepath.Join(installChroot.RootDir(), bootDir)
+	grub2InstallBootDirArg := fmt.Sprintf("%s=%s", bootDirArg, installBootDir)
 
+	err = shell.ExecuteLive(squashErrors, "grub2-install", grub2InstallTargetArg, grub2InstallBootDirArg, bootDevPath)
+	if err != nil {
+		return
+	}
+	installGrub2BootDir := filepath.Join(installChroot.RootDir(), grub2BootDir)
+	err = shell.ExecuteLive(squashErrors, "chmod", "-R", "go-rwx", installGrub2BootDir)
```

Comment:
Question about paths: should `grub2-install` and `chmod` operate on the same directory? What I mean is, for `grub2-install` we're using the value passed into this function through `bootDevPath`. For `chmod` we're using the locally hard-coded `[chroot_root]/boot/grub2`. These two paths are coming from two different sources, so I'd like to make sure that it's acceptable by design, if these paths don't match.

Reply:
`[installroot]/boot/grub2` is where the valuable grub data gets installed to (grub modules, grub config, fonts, etc).